From 529f1ddf4d785ea62fdfc77c62b46cfc611202ac Mon Sep 17 00:00:00 2001
From: Marvin Scholz <epirat07@gmail.com>
Date: Wed, 21 Oct 2026 00:32:03 +0000
Subject: [PATCH] avcodec/mvcdec: zero-initialize color array

Fixes MSAN use-of-uninitialized-value.

Signed-off-by: Marvin Scholz <epirat07@gmail.com>
(cherry picked from commit 3a5eb948e76dc6390e4b3cff7becf84bc492b7af)
---
 libavcodec/mvcdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/mvcdec.c b/libavcodec/mvcdec.c
index 44cab188..a438c8a2 100644
--- a/libavcodec/mvcdec.c
+++ b/libavcodec/mvcdec.c
@@ -138,7 +138,7 @@ static int decode_mvc2(AVCodecContext *avctx, GetByteContext *gb,
                        int linesize, int vflip)
 {
     uint8_t *dst;
-    uint32_t color[128], v[8];
+    uint32_t color[128] = {0}, v[8];
     int w, h, nb_colors, i, x, y, p0, p1, mask;
 
     if (bytestream2_get_bytes_left(gb) < 6)
-- 
2.53.0

